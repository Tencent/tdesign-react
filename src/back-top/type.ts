/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBackTopProps {
  /**
   * 是否绝对定位固定到屏幕右下方
   * @default true
   */
  fixed?: boolean;
  /**
   * 图标
   * @default 'backtop'
   */
  icon?: TNode;
  /**
   * 定位滚动到指定对象
   * @default () => window
   */
  target?: () => HTMLElement;
  /**
   * 文案
   * @default ''
   */
  text?: string;
  /**
   * 预设的样式类型
   * @default round
   */
  theme?: 'default' | 'primary' | 'dark';

  /**
   *  组件的形状
   */
  shape?: 'square' | 'circle';

  /**
   * 组件的大小
   */
  size?: 'small' | 'medium';
}
