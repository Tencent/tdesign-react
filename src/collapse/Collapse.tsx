import React, { useRef } from 'react';
import classnames from 'classnames';
import { TdCollapseProps, CollapsePanelValue, CollapseValue } from './type';
import { StyledProps } from '../common';
import useConfig from '../hooks/useConfig';
import forwardRefWithStatics from '../_util/forwardRefWithStatics';
import useControlled from '../hooks/useControlled';
import CollapsePanel from './CollapsePanel';
import CollapseContext from './CollapseContext';
import { collapseDefaultProps } from './defaultProps';

export interface CollapseProps extends TdCollapseProps, StyledProps {
  children?: React.ReactNode;
}

const Collapse = forwardRefWithStatics(
  (props: CollapseProps, ref: React.Ref<HTMLDivElement>) => {
    const { classPrefix } = useConfig();
    const componentName = `${classPrefix}-collapse`;
    const borderlessClass = `${classPrefix}--border-less`;
    const { defaultExpandAll, disabled, expandIconPlacement, expandOnRowClick, expandIcon, ...rest } = props;
    const { children, className, style, expandMutex, borderless, onChange } = rest;
    const [collapseValue, setCollapseValue] = useControlled(props, 'value', onChange);
    const collapseValues = useRef(collapseValue);

    const updateCollapseValue = (value: CollapsePanelValue) => {
      let newValue: CollapseValue = [].concat(collapseValues.current || []);
      const index = newValue.indexOf(value);
      if (index >= 0) {
        newValue.splice(index, 1);
      } else if (expandMutex) {
        newValue = [value];
      } else {
        newValue.push(value);
      }
      collapseValues.current = [...newValue];
      setCollapseValue(newValue);
    };

    const classes = [
      componentName,
      {
        [borderlessClass]: !!borderless,
      },
      className,
    ];

    const childrenList = React.Children.toArray(children).filter(
      (child: JSX.Element) => child.type.displayName === CollapsePanel.displayName,
    );

    const collapsePanelList = () =>
      childrenList.map((child: React.ReactElement, index: number) => {
        const key = child.key || String(index);
        const childProps = {
          key,
          index: index + 1,
          ...child.props,
        };
        return React.cloneElement(child, childProps);
      });

    return (
      <CollapseContext.Provider
        value={{
          defaultExpandAll,
          disabled,
          expandIconPlacement,
          expandOnRowClick,
          expandIcon,
          updateCollapseValue,
          collapseValue,
        }}
      >
        <div className={classnames(classes)} style={style} ref={ref}>
          {collapsePanelList()}
        </div>
      </CollapseContext.Provider>
    );
  },
  {
    Panel: CollapsePanel,
  },
);

Collapse.displayName = 'Collapse';
Collapse.defaultProps = collapseDefaultProps;

export default Collapse;
