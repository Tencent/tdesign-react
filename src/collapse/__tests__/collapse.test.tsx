import React from 'react';
import { testExamples, render } from '@test/utils';
import Collapse from '../Collapse';
import CollapsePanal from '../CollapsePanel';
import { TdCollapseProps, TdCollapsePanelProps } from '../type';

// 测试组件代码 Example 快照
testExamples(__dirname);

type TestComponentProps = {
  collapse?: TdCollapseProps;
  panel?: TdCollapsePanelProps;
};
const panelContent =
  '这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。';
const CollapseTestComponent = (props: TestComponentProps) => (
  <Collapse {...props?.collapse}>
    <CollapsePanal {...props?.panel}>{panelContent}</CollapsePanal>
  </Collapse>
);

describe('Collapse Unit Test', () => {
  test('create', async () => {
    const { container, queryByText } = render(<CollapseTestComponent />);
    expect(container.firstChild.classList.contains('t-collapse')).toBeTruthy();
    expect(queryByText(panelContent)).toBeInTheDocument();
  });
});
