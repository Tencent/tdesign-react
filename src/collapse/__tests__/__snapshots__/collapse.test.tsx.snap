// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-row"
  >
    <div
      class="t-collapse"
    >
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-fake-arrow--active t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            设置默认展开项
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active appear appear-active"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            自定义折叠面板内容
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              <div
                class="tdesign-demo-block-column"
                style="width: 80%;"
              >
                <div
                  class="t-tag-input t-input__wrap"
                  value=""
                >
                  <div
                    class="t-input t-size-m t-align-left t-is-default t-input--prefix"
                  >
                    <div
                      class="t-input__prefix"
                    >
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        Vue
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        React
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                    </div>
                    <input
                      class="t-input__inner"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            当前折叠面板折叠时，销毁面板内容
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            嵌套使用折叠面板
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              <div
                class="t-collapse"
              >
                <div
                  class="t-collapse-panel"
                >
                  <div
                    class="t-collapse-panel__wrapper"
                  >
                    <div
                      class="t-collapse-panel__header t-is-clickable"
                    >
                      <svg
                        class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
                        fill="none"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                          stroke="black"
                          stroke-opacity="0.9"
                          stroke-width="1.3"
                        />
                      </svg>
                      这是一个折叠标题
                      <div
                        class="t-collapse-panel__header--blank"
                      />
                    </div>
                    <div
                      class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
                      style="height: 0px;"
                    >
                      <div
                        class="t-collapse-panel__content"
                      >
                        这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="t-collapse-panel"
                >
                  <div
                    class="t-collapse-panel__wrapper"
                  >
                    <div
                      class="t-collapse-panel__header t-is-clickable"
                    >
                      <svg
                        class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
                        fill="none"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                          stroke="black"
                          stroke-opacity="0.9"
                          stroke-width="1.3"
                        />
                      </svg>
                      这是一个折叠标题
                      <div
                        class="t-collapse-panel__header--blank"
                      />
                    </div>
                    <div
                      class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
                      style="height: 0px;"
                    >
                      <div
                        class="t-collapse-panel__content"
                      >
                        这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-row"
  >
    <div
      class="t-collapse"
    >
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            折叠后自动销毁
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            自定义折叠面板内容
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              <div
                class="tdesign-demo-block-column"
                style="width: 80%;"
              >
                <div
                  class="t-tag-input t-input__wrap"
                  value=""
                >
                  <div
                    class="t-input t-size-m t-align-left t-is-default t-input--prefix"
                  >
                    <div
                      class="t-input__prefix"
                    >
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        Vue
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        React
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                    </div>
                    <input
                      class="t-input__inner"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="t-radio-group t-size-m t-radio-group__outline"
      >
        <label
          class="t-radio t-is-checked"
        >
          <input
            checked=""
            class="t-radio__former"
            type="radio"
            value="1"
          />
          <span
            class="t-radio__input"
          />
          <span
            class="t-radio__label"
          >
            左边
          </span>
        </label>
        <label
          class="t-radio"
        >
          <input
            class="t-radio__former"
            type="radio"
            value="2"
          />
          <span
            class="t-radio__input"
          />
          <span
            class="t-radio__label"
          >
            右边
          </span>
        </label>
        <label
          class="t-radio"
        >
          <input
            class="t-radio__former"
            type="radio"
            value="3"
          />
          <span
            class="t-radio__input"
          />
          <span
            class="t-radio__label"
          >
            不展示
          </span>
        </label>
      </div>
      <div>
        <label
          class="t-checkbox"
        >
          <input
            class="t-checkbox__former"
            type="checkbox"
            value=""
          />
          <span
            class="t-checkbox__input"
          />
          <span
            class="t-checkbox__label"
          >
            仅图标响应点击
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`mutex.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-row"
  >
    <div
      class="t-collapse"
    >
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            折叠后自动销毁
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            自定义折叠面板内容
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              <div
                class="tdesign-demo-block-column"
                style="width: 80%;"
              >
                <div
                  class="t-tag-input t-input__wrap"
                  value=""
                >
                  <div
                    class="t-input t-size-m t-align-left t-is-default t-input--prefix"
                  >
                    <div
                      class="t-input__prefix"
                    >
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        Vue
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        React
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                    </div>
                    <input
                      class="t-input__inner"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`other.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-row"
  >
    <div
      class="t-collapse"
    >
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            折叠后自动销毁
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            自定义折叠面板内容
            <div
              class="t-collapse-panel__header--blank"
            />
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              <div
                class="tdesign-demo-block-column"
                style="width: 80%;"
              >
                <div
                  class="t-tag-input t-input__wrap"
                  value=""
                >
                  <div
                    class="t-input t-size-m t-align-left t-is-default t-input--prefix"
                  >
                    <div
                      class="t-input__prefix"
                    >
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        Vue
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                      <span
                        class="t-tag t-tag--default t-tag--dark t-tag--medium t-tag--square"
                      >
                        React
                        <svg
                          class="t-icon t-icon-close t-tag__icon-close"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </span>
                    </div>
                    <input
                      class="t-input__inner"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          type="checkbox"
          value=""
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          禁止全部
        </span>
      </label>
      <label
        class="t-checkbox"
      >
        <input
          class="t-checkbox__former"
          type="checkbox"
          value=""
        />
        <span
          class="t-checkbox__input"
        />
        <span
          class="t-checkbox__label"
        >
          无边框模式
        </span>
      </label>
      <div
        style="margin-top: 10px;"
      >
        当前展开的Collapse Panel: 
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rightSlot.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-row"
  >
    <div
      class="t-collapse"
    >
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
            <button
              class="t-button t-button--theme-primary t-button--variant-base t-size-s"
              type="button"
            >
              <span
                class="t-button__text"
              >
                操作
              </span>
            </button>
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
            <button
              class="t-button t-button--theme-primary t-button--variant-base t-size-s"
              type="button"
            >
              <span
                class="t-button__text"
              >
                操作
              </span>
            </button>
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-collapse-panel"
      >
        <div
          class="t-collapse-panel__wrapper"
        >
          <div
            class="t-collapse-panel__header t-is-clickable"
          >
            <svg
              class="t-fake-arrow t-collapse-panel__icon t-collapse-panel__icon--left"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
                stroke="black"
                stroke-opacity="0.9"
                stroke-width="1.3"
              />
            </svg>
            这是一个折叠标题
            <div
              class="t-collapse-panel__header--blank"
            />
            <div
              style="display: flex; align-items: center;"
            >
              <label
                class="t-checkbox"
              >
                <input
                  class="t-checkbox__former"
                  type="checkbox"
                  value=""
                />
                <span
                  class="t-checkbox__input"
                />
                <span
                  class="t-checkbox__label"
                >
                  是否禁止
                </span>
              </label>
              <button
                class="t-button t-button--theme-primary t-button--variant-base t-size-s"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作
                </span>
              </button>
            </div>
          </div>
          <div
            class="t-collapse-panel__body t-slide-down-enter-active t-collapse-panel__body--collapsed"
            style="height: 0px;"
          >
            <div
              class="t-collapse-panel__content"
            >
              这部分是每个折叠面板折叠或展开的内容，可根据不同业务或用户的使用诉求，进行自定义填充。可以是纯文本、图文、子列表等内容形式。
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
